semgrep-core/parsing/Parse_target.ml

```diff
@@ -280,7 +280,6 @@ let parse_and_resolve_name_use_pfff_or_treesitter lang file =
   Naming_AST.resolve lang ast;
   Constant_propagation.propagate_basic lang ast;
   Constant_propagation.propagate_dataflow ast;
-  (* TODO:  Caching.prepare_target ast; commented because of test regressions *)
```

Comment:
No need for prepare_target here anymore?

Reply:
That's correct. It was in charge of numbering the node IDs, but now it's done at the time of creating the stmt node, rather than mutating the field later. The motivation was the ppx_hash extension which refuses to run on mutable fields.